### DISCUSSION TITLE
Assign keybinding to snippet

### DISCUSSION THREAD

**Discussion by ghost** [orbitalquark/textadept#338] [Q&A] [answered]
Hi. Recently discovered Textadept. Thanks much.
New to Textadept and Lua. Two new great adventures, Yay.

I have a snippet that I use quite frequently in html and plain text languages.
Choosing Tools > Snippets > Insert Snippet and then choosing from a list is
cumbersome.

`snip['tag'] = '<%1%2>%<selected_text></%1>'`

I would like to use ``ctrl+` `` to insert this snippet.

I try
``keys['ctrl+`'] = textadept.snippets.insert('<%1%2>%<selected_text></%1>')``
in my init.lua file, but that brings an error buffer when I start Text adept.

The snippet works fine when I put it in html and text language modules.

Any help will be greatly appreciated.

**Comment by orbitalquark** [comment, marked as answer]
Try this:

     keys['ctrl+`'] = function() 
textadept.snippets.insert('<%1%2>%<selected_text></%1>') end

You want to assign to the key a function to call, rather than calling a 
function (on init) and assigning its result to the key.

**Reply by ghost** [reply]
Got it.
Lots to learn.
Awesome response time! Thanks much.
